File: 7_semester/МСПИСИТ/34/backend/src/main/kotlin/suai/vladcto/services/conference/ConferenceRequest.kt
package suai.vladcto.services.conference

import kotlinx.serialization.Serializable

@Serializable
data class ConferenceRequest(
    val name: String,
    val cityId: Int,
    val topicId: Int,
    val startDate: String,
    val endDate: String,
    val description: String,
    val contacts: String,
    val imageUrl: String,
)

File: 7_semester/МСПИСИТ/34/frontend/conference_app/lib/features/conferences/data/models/conference_request.dart

import 'package:freezed_annotation/freezed_annotation.dart';

part 'conference_request.freezed.dart';
part 'conference_request.g.dart';

@freezed
class ConferenceRequest with _$ConferenceRequest {
  const factory ConferenceRequest({
    required String name,
    required int cityId,
    required int topicId,
    required String startDate,
    required String endDate,
    required String description,
    required String contacts,
    required String imageUrl,
  }) = _ConferenceRequest;

  factory ConferenceRequest.fromJson(Map<String, dynamic> json) => _$ConferenceRequestFromJson(json);
}
